lib/runtime/adapter-read.js

```diff
@@ -114,6 +115,15 @@ class AdapterRead {
         return errors.runtimeError(code, info);
     }
 
+    // Helper to obtain a throttled function to emit warnings.
+    throttledWarn(code, info, interval) {
+        interval = interval || 1000;
+        info = info || {};
+        return _.throttle(() => {
+            this.trigger('warning', this.runtimeError(code, info));
+        }, interval);
+    }
```

Comment:
We already have similar logic in `periodic_fanin` (also for triggering an error only once), maybe it would be worth it moving this functionality into a common module, say `throttled_errors` / support it in `errors` directly? @dmajda thoughts?

In fanin, the interval value is related to `interval` option - would it make sense to set the default to the value of `every` in adapters?